### DISCUSSION TITLE
Docker container for headless server

### DISCUSSION THREAD

**Discussion by EspenRoth** [NebulaModTeam/nebula#591] [General]
Is anyone currently working on a docker container that could be used to jumpstart a headless DSP multiplayer server? Would you accept a PR that does it?

**Comment by scottbot95** [comment]
There may potential legal issues given that even the headless server is just a mod for the base game so they would essentially be vending a pirated copy of the game by providing a docker image for the headless server.

I could imagine this being a thing, but I think it would require the server to be it's own executable that is completely decoupled from the base game. And that's probably not an insignificant amount of effort.

**Reply by PhantomGamers** [reply]
Theoretically someone could make a docker container where you just have to pass it the directory of the game I think. I'm not aware of anyone working on such a project though.

**Comment by adaliszk** [comment]
I would not expect a docker image until we get a dedicated Linux build, no matter what kind of extra layers need to be done for legality. Most Windows-based hosting solutions just run the server directly and don't make an effort to containerize, however, once Linux hosting is available without using wine then it is usually done.

**Comment by PhantomGamers** [comment]
https://github.com/WhyKickAmooCow/dsp-docker-server
